identity/src/main/java/com/android/identity/credential/NameSpacedData.kt

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.credential
+
+import co.nstant.`in`.cbor.CborBuilder
+import co.nstant.`in`.cbor.builder.MapBuilder
+import co.nstant.`in`.cbor.model.ByteString
+import co.nstant.`in`.cbor.model.DataItem
+import co.nstant.`in`.cbor.model.Map
+import co.nstant.`in`.cbor.model.Tag
+import co.nstant.`in`.cbor.model.UnicodeString
+import com.android.identity.internal.Util
+
+/**
+ * Key/value pairs, organized by name space.
+ *
+ * This class implements a data model which consists of a series of name spaces
+ * (identified by a string such as *org.iso.18013.5.1*) where each name space
+ * contains a number of key/value pairs where keys are strings and values are
+ * [CBOR](https://datatracker.ietf.org/doc/html/rfc8949) values.
+ *
+ * While this happens to be similar to the mDL/MDOC data model used in
+ * [ISO/IEC 18013-5:2021](https://www.iso.org/standard/69084.html),
+ * it's flexible enough to be used to store credential data for any kind of
+ * credential.
+ *
+ * This type is immutable.
+ */
+class NameSpacedData {
+    private var map: LinkedHashMap<String, LinkedHashMap<String, ByteArray>>
+
+    internal constructor() {
+        map = LinkedHashMap()
+    }
+
+    private constructor(map: LinkedHashMap<String, LinkedHashMap<String, ByteArray>>) {
+        this.map = map
+    }
```

Comment:
In Kotlin DSL, these 2 constructors don't need to exist since the `internal` constructor resorts to instantiating an empty map and the private one is called from the builder... these along with the class can be rewritten to 

```
class NameSpacedData private constructor(private val map: MutableMap<String, MutableMap<String, ByteArray>>) {
```